### PR TITLE
Fix rpc-go-server to not use hard coded types package

### PR DESCRIPTION
**Pull request by kklas**
Made required fixes so that rpc-go-server generates the correct types when used with the `-types` option.

Also added fixtures to help with testing.

### CODE REVIEW

File: cmd/rpc-go-server/fixtures/todo_types_gen.go.fix [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  "net/http"
+
+  "github.com/apex/rpc"
+  "github.com/apex/log"
+
+  "github.com/apex/rpc/cmd/rpc-go-server/api"
+)
+
+// ServeHTTP implementation.
+func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```
Comment (tj) [inline cmd/rpc-go-server/fixtures/todo_types_gen.go.fix:16]: I haven't used them in a while but I think Go special-cases `testdata` directories, all the tooling avoids them I believe, we could do `./testdata/todo_types_gen.go` I think, not a big deal either way

**Comment by tj** [comment]
Looks good to me thanks! I added a few notes for what I think we could change later, but for now looks good!

File: cmd/rpc-go-server/Makefile [hunk 2]
```diff
@@ -0,0 +1,7 @@
+gen-fixtures:
+	@echo "==> generate fixtures/todo_no_types_gen.go.fix"
+	@go run main.go -schema ../../examples/todo/schema.json > fixtures/todo_no_types_gen.go.fix
+	@echo "==> generate fixtures/todo_types_gen.go.fix"
+	@go run main.go -types github.com/apex/rpc/cmd/rpc-go-server/api -schema ../../examples/todo/schema.json > fixtures/todo_types_gen.go.fix
+
```
[the remark this answers is not in the document]
Comment (kklas) [reply cmd/rpc-go-server/Makefile:6]: Yea I saw this, but it's a separate change. Wanted to keep this simple.

File: cmd/rpc-go-server/fixtures/todo_types_gen.go.fix (hunk 1 above)
Comment (kklas) [reply cmd/rpc-go-server/fixtures/todo_types_gen.go.fix:16]: Ahh, didn't know this. Will keep this in mind in the future.

**Pull request merged**